core/src/main/scala/kafka/log/remote/RemoteLogIndexEntry.scala

```diff
@@ -0,0 +1,229 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package kafka.log.remote
+
+import java.io.InputStream
+import java.nio.{BufferUnderflowException, ByteBuffer}
+import java.nio.channels.FileChannel
+import java.util
+import java.util.zip.CRC32
+
+import kafka.common.KafkaException
+import org.apache.kafka.common.utils.Utils
+import kafka.utils.Logging
+
+/**
+ * Entry representation in a remote log index
+ *
+ * @param magic          magic version of protocol
+ * @param crc            checksum value of the entry
+ * @param firstOffset    offset value of the first record for this entry stored at respective { @link #rdi}
+ * @param lastOffset     offset value of the last record for this entry stored at respective { @link #rdi}
+ * @param firstTimeStamp timestamp value of the first record for this entry stored at respective { @link #rdi}
+ * @param lastTimeStamp  timestamp value of the last record for this entry stored at respective { @link #rdi}
+ * @param dataLength     length of the data stored in remote tier at rdi.
+ * @param rdi            bytes value of rdi.
+ */
+case class RemoteLogIndexEntry(magic: Short, crc: Int, firstOffset: Long, lastOffset: Long,
```

Comment:
Out of curiosity, what would happen in case of corrupted `RemoteLogIndexEntry` with an invalid CRC code? Here it seems it is used as part of the object's identity - do you plan to perform explicit check to detect potential inconsistencies?